MobileWallet/Screens/Home/TransactionHistory/TransactionsTableViewController.swift

```diff
@@ -127,12 +135,92 @@ class TransactionsTableViewController: UITableViewController {
-            sender.endRefreshing()
+    private func showRefreshingSpinner(_ show: Bool) {
+        refreshTransactionControl.attributedTitle = nil
+
+        if show {
+            refreshTransactionControl.tintColor = .systemGray
+            refreshTransactionControl.subviews.first?.alpha = 1
+        } else {
+            UIView.animate(withDuration: 0.25, animations: { [weak self] in
+                guard let self = self else { return }
+                self.refreshTransactionControl.tintColor = .clear
+                self.refreshTransactionControl.subviews.first?.alpha = 0
+            }) { _ in
+                //Switch off the pull to refresh part
+                self.refreshTransactionControl.endRefreshing()
+            }
+        }
+    }
+
+    private func listenForRefreshUpdates() {
+        self.animatedRefresher.setupView(.loading)
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5, execute: { [weak self] in
+            guard let self = self else { return }
+            //We know where refreshing if this is the first load
+            self.beginRefreshing()
         })
+
+        //After first load if they bring the app into the foreground listen for the connection starting
+        TariEventBus.onMainThread(self, eventType: .torPortsOpened) { [weak self] (_) in
+            guard let self = self else { return }
+            self.beginRefreshing()
+        }
+
+        //Listen for when to stop refreshing
+        TariEventBus.onMainThread(self, eventType: .baseNodeSyncComplete) { [weak self] (result) in
+            guard let self = self else { return }
+
+            if let success: Bool = result?.object as? Bool {
+                if success == true {
+                    self.endRefreshingWithSuccess()
+                } else {
+                    //TODO base node always fails the first few times. When that stabalises we can show an error here.
+                }
+            }
+        }
+    }
+
+    private func beginRefreshing() {
+        showRefreshingSpinner(false)
+
+        guard !isRefreshing else {
+            return
+        }
+        isRefreshing = true
+
+        animatedRefresher.updateState(.loading)
+        animatedRefresher.animateIn()
+
+        try? TariLib.shared.tariWallet?.syncBaseNode()
+    }
+
+    private func endRefreshingWithSuccess() {
+        guard isRefreshing else {
+            return
+        }
+
+        self.animatedRefresher.updateState(.success)
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 2, execute: { [weak self] in
+            guard let self = self else { return }
+
+            self.animatedRefresher.animateOut { [weak self] in
+                guard let self = self else { return }
+
+                self.isRefreshing = false
+
+                DispatchQueue.main.asyncAfter(deadline: .now() + 1, execute: { [weak self] in
+                    guard let self = self else { return }
+                    self.showRefreshingSpinner(true)
```

Comment:
From the previous `DispatchQueue.main.asyncAfter` up to this one reads to me like: the `animatedRefresher` animates out after 2 seconds of the calling of `endRefreshingWithSuccess`, then 1 second after the completion of `animateOut` the refresh spinner shows again. Looking at the name of the method I wouldn't expect the refreshing spinner show again. It's probably doing it right but just wanted to note how it reads.

Reply:
I see, I think I need to rename the function. By "show" I meant don't make it transparent anymore. When I got pull to refresh working I needed to phase out that spinner after it's pulled else there are 2 spinners visible.